### DISCUSSION TITLE
对于博士论文来说，最佳实践是否是：各章节单独一个tex文件，最后 \include 到一起

### DISCUSSION THREAD

**Discussion by runck** [TheNetAdmin/zjuthesis#258] [General]
由于博士论文较长，有时只修改一两页，如果重新编译整个文件感觉不太优雅。

搜索时发现可以使用 \include 来分离各章节的编译，但看到 zjuthesis 中各章节是放在一个tex中的，不知这种分成单文件用 \include 的方式是否有效，或者哪种是这类论文使用时的最佳实践？

谢谢！

**Comment by xfz329** [comment]
反正我是按章节拆开成独立文件的。得出的经验教训是，千万不要用chapter1.tex、chapter2.tex这类名字。新增加章节时太容易引起混乱了。

**Reply by runck** [reply]
了解，多谢指教！

**Comment by TheNetAdmin** [comment]
一般来讲 `\include` 可以避免一些重复编译，但并不是绝对的。比如你修改了第二章，其中新增了一页，那么后续章节都会因为新增一页而重新编译。

当前模板使用了 `\input` 而非 `\include`，主要原因是 `\include` 会 clear 其文件最后一页，但这个行为并不适用所有情况。我建议你可以将 chapter 级别的文件使用 `\include`，应该可以适当地减少重复编译。

也就是说在 `body/graduate/content.tex` 文件里使用 `\include` 来插入子文件。

不过我在写自己的毕业论文时感觉 `\include` 的帮助有限，很多情况下还是会全文重新编译。

**Reply by runck** [reply]
谢谢建议！

**Comment by ToughJ** [comment]
顺便问一下，使用\input 或者\include后  TexStudio的text-code之间的跳转变得不正常了，应该如何解决呢？

**Reply by xfz329** [reply]
没用过，我一直只用vscode写

**Comment by genema** [comment]
意义比较有限

**Comment by Chenyanlu99** [comment]
分开写之后，怎么对全文进行字数统计呢？

**Reply by xfz329** [reply]
只要你在zjuthesis.tex 中正确添加了多个tex文件
![image](https://github.com/user-attachments/assets/5a8567da-6fff-4ac1-9d01-63796ecb7e15)
运行\script\utils\word_count.sh 即可统计字数
